### PR TITLE
Adding script to collect MAC addresses and do a basic pre-install check

### PR DESCRIPTION
**Pull request by LawrenceSpracklen**

### CODE REVIEW

**Comment by LawrenceSpracklen** [comment]
@kevinmtrowbridge @haozou @nkapinos 

First cut at python MAC script. Also, does some of the checks that the old go exe used to do. I give it a try on SLES 11, 12 and CentOS 6 and 7.

File: packaging/setup/precheck_lib/mac_check.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+#
+# CHECK NETWORKING
+#
+
+#Get MAC Address
+print "\n===> Determine MAC address"
+try:
+    print "System MAC address is : %x" % (uuid.getnode())
+except:
```
Comment (kevinmtrowbridge) [inline packaging/setup/precheck_lib/mac_check.py]: Hi @LawrenceSpracklen: I ran this on my mac -- but I don't think it's a platform issue ... it returned:

```
===> Determine MAC address
System MAC address is : 6c4008b2f24e
```

The license will require the bits to be separated by colons, for example how the `ifconfig` command returns it:

```
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 6c:40:08:b2:f2:4e
```

In this stack overflow post, this same problem is discussed:
http://stackoverflow.com/questions/159137/getting-mac-address

They suggest this extra piece of python:

> Or `':'.join(("%012X" % mac)[i:i+2] for i in range(0, 12, 2))` for an uppercase MAC with each byte separated by a colon. – tomlogic Nov 10 '14 at 17:21

**Comment by LawrenceSpracklen** [comment]
@kevinmtrowbridge Integrated suggested fix.

**Comment by haozou** [comment]
There are some health check duplicate with the one under packaging/setup/health_lib. Maybe you can put some of your checks under that. I can talk with you in person whenever you are available.

**Comment by LawrenceSpracklen** [comment]
Sure -- let's sync tomorrow.

**Comment by kevinmtrowbridge** [comment]
So, I'm trying to think through how this will go when we work with a customer ... in the install docs that we give to our support people, we'll say something like:

1) Install chorus
2) After the installation, do not start chorus
2) Run `/usr/local/chorus/current/packaging/setup/precheck_lib/mac_check.py`
3) Email us the output
4) Wait ~1-24h
5) We send you a license, place that here: `/usr/local/chorus/shared/chorus.license`
6) Start chorus

Why put the precheck_lib in the chorus repo?  We probably won't run this script as a part of actually running chorus, only before -- which is strange -- install chorus so you can run a script called "precheck"?  We could store this anywhere, in a Github gist or, we could create a page on the licensing console which allows folks to submit requests for licenses ... which would have the instructions:

1) Go to http://licensing.alpinenow.com/request_license
2) Follow instructions (which tell you to download & run script on the machine where you want to install alpine)
3) Submit output of the script to the licensing console as a request for a new license
4) You get your license
5) Install chorus
6) Copy license into place
7) Run chorus

**Comment by kevinmtrowbridge** [comment]
Ok one final thought: if we are going to store precheck_lib in the chorus repo, we should add a README.md file to this directory just explaining what it's for, when we anticipate running it, and who to contact if you have questions about it.

**Comment by LawrenceSpracklen** [comment]
Agreed. I think this is more just a place to hold it -- so maybe its the wrong place. This script will be provided to the customer to run on a system on which they intended to subsequently install chorus. This script will not only mine the MAC address so was can spin a license, but will also do a basic sanity check of the system to ensure that it has sufficient resources to act as a good host. The hope is that we can find problems prior to the chorus install event.

**Comment by kevinmtrowbridge** [comment]
The chorus repo is a "kitchen sink" and, one of the things I am trying to do is to stop that.  Code checked into the chorus repo is going to be constantly pawed over and altered by the forces of entropy.

We could store it in a separate Github repo, secret gist, in the Jira support portal somewhere ... in the licensing console or linked to by the licensing console ... perhaps best if it lives with the support folks, not with engineering?  Just throwing out ideas.  Because I am trying to delete / cleanup the chorus repo so it becomes less overwhelming.

**Comment by LawrenceSpracklen** [comment]
Moved code to a new Support repo.

**Pull request closed without merging**